system/mp/mpcomm.cpp

```diff
@@ -1889,6 +1914,14 @@ int CMPConnectThread::run()
         }
         if (sock) {
             try {
+                if (soMaxCheck)
+                {
+                    if (numMPAccepts++ >= soMaxConn/2)
```

Comment:
It may be possible to look at connect() failures from the other side and give a warning as well, but that can be added as another task.
Ultimately the mp connection should launch or pass to another thread to complete the handshake so that accept()s happen very quickly and there is little chance for the backlog to grow.